### ISSUE TITLE
Binance Smart Chain - BSC support

### ISSUE THREAD

**Issue by feraudet** [emeraldpay/dshackle#146]
Hello,
Could it be possible to add BSC support ? 
I'm not a java/groovy aficionados but as far as I can see it's seems possible by adding some lines, I am wrong ?
What is your thought ?
Cyril

**Comment by quickchase** [comment]
If you copy #103 with the right chain info for BSC, it should work...

**Comment by splix** [comment]
Yes, as @quickchase said, at this moment the idea is to use it just use it as `ethereum`. Consider it as a _generic ethereum_ configuration. It works. But the limitation of that approach it that you'll be limited with only one ethereum chain withing a single Dshackle instance. I mean you can have multiple upstreams, just make sure all of them are on same chain. Don't mix Ethereum Mainnet and Binance Mainnet on same instance.

**Comment by feraudet** [comment]
> Yes, as @quickchase said, at this moment the idea is to use it just use it as `ethereum`. Consider it as a _generic ethereum_ configuration. It works. But the limitation of that approach it that you'll be limited with only one ethereum chain withing a single Dshackle instance. I mean you can have multiple upstreams, just make sure all of them are on same chain. Don't mix Ethereum Mainnet and Binance Mainnet on same instance.

It's what I did but it constraint me to have a dedicated instance for BSC and send stats tagged as "ETH" in prom :/

**Comment by feraudet** [comment]
> If you copy #103 with the right chain info for BSC, it should work...

Code seems to have changed a lot since this PR. But I will try by my own

**Comment by splix** [comment]
yes, right, that's a limitation. In all logs and metrics it going to be just ETH, that's why I called it "generic ethereum"

**Comment by nforced** [comment]
This will be a huge leap for the project, dapps use lots of networks and BSC is a big one that we need to utilize in the project. Having a separate instance just because of ETH vs BSC network is unwanted trouble that seems not a viable solution to the problem.

**Comment by Zorlin** [comment]
@splix How do you use BSC as generic ethereum? Isn't the chainID different?
